lib/mongodb.js

```diff
@@ -573,18 +555,34 @@ MongoDB.prototype.create = function(modelName, data, options, callback) {
   if (self.debug) {
     debug('create', modelName, data);
   }
-  let idValue = self.getIdValue(modelName, data);
+
+  const idValue = self.getIdValue(modelName, data);
   const idName = self.idName(modelName);
+  const propDef = self.getPropertyDefinition(modelName, idName);
+
+  if (hasDataType('objectid', propDef) && !propDef.generated && idValue) {
+    if (!ObjectIdValueRegex.test(idValue)) {
+      const err = new Error('Invalid ObjectId string.');
+      return callback(err);
+    }
+  } else if (propDef.generated && idValue) {
+    const err = new Error('Cannot specify value for autogenerated id.');
```

Comment:
I am a bit confused. Which case are we trying to catch here? Are we trying to reject if uses want to have auto-generated number/non-object-like string type id?

Reply:
I am removing id checks for now, it is not compatible with `replaceOrCreate`.